src/network/netdev/wireguard.c

```diff
@@ -213,6 +213,7 @@ static int on_resolve_retry(sd_event_source *s, usec_t usec, void *userdata) {
 
         w->resolve_retry_event_source = sd_event_source_unref(w->resolve_retry_event_source);
 
+        assert(!w->unresolved_endpoints);
```

Comment:
I double checked but did not found a case where this could happen, since `on_resolve_retry` is only scheduled
when all previous endpoints have been resolved. It is still a good idea though to have this here.